lib/rubycritic/cli/application.rb

```diff
@@ -14,12 +14,29 @@ def initialize(argv)
 
       def execute
         parsed_options = @options.parse
-        ::Rubycritic::CommandFactory.create(parsed_options).execute
+        result = ::Rubycritic::CommandFactory.create(parsed_options).execute
+
+        return STATUS_ERROR unless satisfy_minimum_score?(result, parsed_options)
+
         STATUS_SUCCESS
       rescue OptionParser::InvalidOption => error
         $stderr.puts "Error: #{error}"
         STATUS_ERROR
       end
+
+      private
+
+      def satisfy_minimum_score?(result, options)
+        return true unless result.is_a?(Hash) && result[:score]
```

Comment:
Let's not do this. This is 1.) extremely obscure (no innocent reader will understand why you are doing this at first glance) and 2.) makes heavy use of type checking, something that is smelly per se in a dynamically typed language like Ruby.
We should test here that we satisfy the minimum-score, nothing else. The check that this option is set should happen above, in "execute".